doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,195 @@
+# Developer driven evolution
+To add a convenience API that takes and returns strongly-typed input and output models instead of raw JSON, steps are:
+1. [Create the convenience method with new signature](#create-the-convenience-method-with-new-signature)
+- [Replace input parameters](#replace-input-parameters)
+- [Replace output value](#replace-output-value)
+- [Rename method if needed](#rename-method-if-needed)
+2. [Create the models](#create-the-models)
+3. [Implement the method](#implement-the-method)
+- [Implement GET method](#implement-get-method)
+- [Implement POST method](#implement-post-method)
+- [Implement GET paging method](#implement-get-paging-method)
+
+## Create the convenience method with new signature
+You should first decide how the convenience method grows up from protocol method. Below is an example of your generated protocol method.
+```C#
+public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
+```
+You should do the following steps to create the convenience method signature:
+### Replace input parameters
+- Replace the parameter `RequestContext` with an optional `CancellationToken` with default value.
+- If there is an input parameter `RequestContent`, replace it with a wanted input type (e.g., `MetricFeedback`).
```

Comment:
How can we generate the model for 'MetricFeedback'?

Reply:
And what is the folder structure for the manual code to avoid them to be overwritten when we call codegen again?

Reply:
> How can we generate the model for 'MetricFeedback'?

How about we put section [Create the models](#create-the-models) as the first step?

Reply:
It is better to tell what 'MetricFeedback' is before referencing this model.